Task 4/Task 4.1/FIleManagementSystem/Dependency/DependencyResolver.cs

```diff
@@ -0,0 +1,22 @@
+﻿using FIleManagementSystem.Interfaces;
+using FIleManagementSystem.Logic;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FIleManagementSystem.Dependency
+{
+    public static class DependencyResolver
+    {
+        private static IBackupLogic _backupLogic;
+        public static IBackupLogic BackupLogic
+            => _backupLogic is null ? _backupLogic = new BackupLogic() : _backupLogic;
```

Comment:
```suggestion
        private static IBackupLogic _backupLogic;
        public static IBackupLogic BackupLogic => _backupLogic ??= new BackupLogic();
```

You can also read about Lazy<T> class - it offers thread-safe lazy initialization